### PR TITLE
Add 'Dont_override_old_maneuvers' flag to maneuvers

### PR DESCRIPTION
**Pull request by Baezon**
Maneuvers normally clear any existing maneuvers before taking effect, with this flag it will not do that. Though it does refresh the duration of the old maneuver, which is unideal, this allows for independent sexp/scripting control over lateral and rotational maneuvers at the same time, which is incredibly useful.

### CODE REVIEW

**Review by Goober5000 — APPROVED** [review]

**Review by asarium — APPROVED** [review]

**Comment by Goober5000** [comment]
> Though it does refresh the duration of the old maneuver

You may want to separate the timestamps for lateral and rotational maneuvers, which would solve this problem.

**Comment by Baezon** [comment]
Ah that's true, that would be a fairly effective but lightweight way to do it, although it would be more awkward to describe in the sexp doc...

**Comment by Baezon** [comment]
The code is a bit more cumbersome this way, but it would definitely be more convenient for the modder.

File: code/ai/aicode.cpp [hunk 1]
```diff
@@ -14227,8 +14227,28 @@ static void ai_control_info_check(ai_info *aip, ship_info *sip, physics_info *pi
 	if (aip->ai_override_flags.none_set())
 		return;
 
-	if (!aip->ai_override_flags[AI::Maneuver_Override_Flags::Never_expire] && timestamp_elapsed(aip->ai_override_timestamp))
+	bool no_lat = false;
+	bool no_rot = false;
+
+	if ((!aip->ai_override_flags[AI::Maneuver_Override_Flags::Lateral_never_expire] && timestamp_elapsed(aip->ai_override_lat_timestamp)))
```
Comment (Goober5000) [inline code/ai/aicode.cpp:14233]: Any reason you have an extra set of parentheses here?

**Review by Goober5000 — APPROVED** [review]
New code looks good.  There's one comment that's more of a nitpick than anything.

File: code/ai/aicode.cpp (hunk 1 above)
Comment (Baezon) [reply code/ai/aicode.cpp:14233]: ah whoops, used to be part of a bigger expression

**Pull request merged**